yarpl/include/yarpl/observable/Observer.h

```diff
@@ -42,11 +43,21 @@ class Observer : public virtual Refcounted {
 
   virtual void onNext(T) = 0;
 
+  bool isUnsubscribed() const {
+    CHECK(subscription_);
```

Comment:
Why do we call CHECK(), is subscription_ is null, shouldn't we just return false?

Reply:
Good suggestion, but thinking about the meaning of isUnsubscribed when the subscriber hasn't subscribed yet (subscription_ == null), it feels wrong.
Reactive streams protocol says when we call onComplete or onError, the subscription object is no longer valid so I would argue that it also makes no sense to call isUnsubscribed.
Would you agree with that?

Reply:
Can we call this `isSubscribed` so we're writing `if(obj->isSubscribed())` rather than the double-negative `if(!obj->isUnsubscribed())`?